src/Stack/Setup.hs

```diff
@@ -1226,7 +1228,29 @@ getGhcBuilds = do
     config <- view configL
     case configGHCBuild config of
         Just ghcBuild -> pure [ghcBuild]
-        Nothing -> determineGhcBuild
+        Nothing -> do
+          mOsReleaseResult <- liftIO OsRelease.parseOsRelease
+          case mOsReleaseResult of
+            Nothing -> determineGhcBuild
+            Just OsRelease.OsReleaseResult{OsRelease.osRelease=osRelease} -> do
+              let OsRelease.OsRelease{OsRelease.id=osId, OsRelease.id_like=osIdLike} = osRelease
```

Comment:
Still doesn't utilize `name`.

Reply:
See my previous comment above, I don't understand why it is necessary to use NAME. Do you know of any Debian derivatives that distributions that don't have ID but do have NAME?

Reply:
No. I'm just saying that none of those keys are mandatory as per the spec, so the logical thing to do is consider all of them.

Reply:
In which case would we fall back from `ID` though? When it has the value `linux`? It is not a `Maybe` in the os-release library.